packages/next-auth/src/core/routes/session.ts

```diff
@@ -71,7 +71,7 @@ export default async function session(
       const newToken = await jwt.encode({
         ...jwt,
         token,
-        maxAge: options.session.maxAge,
+        maxAge: options.jwt.maxAge ?? options.session.maxAge,
```

Comment:
```suggestion
        maxAge: options.jwt.maxAge,
```

Because of https://github.com/nextauthjs/next-auth/pull/4901/files#r918432349, this should now be safe.